<launch>

    <arg name="odom_topic" default="/dragonfly67/quadrotor_ukf/control_odom"/>
    <arg name="robot_name" default="robot0"/>
    <arg name="robot_ID" default="0"/>

    <node name="rgb_sem_segmentation" pkg="object_modeller" type="detect_open_vocab.py" output="screen" ns="$(arg robot_name)">

        <param name="~sim" value="false"/>
        <param name="~desired_rate" value="2"/>
        <param name="~confidence_threshold" value="0.4"/>
        <param name="~rgb_topic" value="camera/color/image_raw/"/>
        <param name="~depth_topic" value="camera/depth/image_rect_raw"/>         
        <param name="~aligned_depth_topic" value="camera/aligned_depth_to_color/image_raw"/>
        <param name="~odom_topic" value="$(arg odom_topic)"/>
        <param name="~sync_odom_measurements" value="true"/>
        <param name="~sync_pc_odom_topic" value="/robot$(arg robot_ID)/sem_detection/sync_pc_odom"/>
        <param name="~pc_topic" value="/robot$(arg robot_ID)/sem_detection/pointcloud"/>



        <!-- For scarab -->
        <!-- <param name="~fx" value="390.97088623046875" />
        <param name="~fy" value="390.970886230468" />
        <param name="~cx" value="324.1226806640625" />
        <param name="~cy" value="243.95254516601562" /> -->

        <!-- For f250 435i -->
        <!-- <param name="~fx" value="603.7166748046875" />
        <param name="~fy" value="603.9064331054688" />
        <param name="~cx" value="314.62518310546875" />
        <param name="~cy" value="244.9166717529297" />
        <param name="~k_depth_scaling_factor" value="1000.0" /> -->
        
        <!-- For f250 455 -->
        <param name="~fx" value="390.97088623046875" />
        <param name="~fy" value="390.970886230468" />
        <param name="~cx" value="324.1226806640625" />
        <param name="~cy" value="243.95254516601562" />
        <param name="~k_depth_scaling_factor" value="1000.0" /> 

    </node>

</launch>
